File: schemas/recipe-cooking/convex/recipe_favorites.ts
// recipe_favorites: Tracks which users have favorited which recipes.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const recipeFavorites = defineTable({
  recipeId: v.id("recipes"),
  userId: v.id("users"),
})
  .index("by_recipe_id_and_user_id", ["recipeId", "userId"])
  .index("by_user_id", ["userId"]);

File: schemas/recipe-cooking/spacetimedb/recipe_favorites.rs
// recipe_favorites: Tracks which users have favorited which recipes.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

// Composite unique(recipe_id, user_id) — enforce in reducer logic

#[spacetimedb::table(name = recipe_favorites, public)]
pub struct RecipeFavorite {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub recipe_id: String, // FK → recipes.id (cascade delete)

    #[index(btree)]
    pub user_id: String, // FK → users.id (cascade delete)

    pub created_at: Timestamp,
}
